### DISCUSSION TITLE
TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer...

### DISCUSSION THREAD

**Discussion by eldar** [danny-avila/LibreChat#4744] [Troubleshooting]
### What happened?

I followed the installation instructions which worked for me before, but I cannot access the `localhost:3080`. Docker logs reveal the following error:

```
TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined
```

### Steps to Reproduce

1. Install Docker and Docker compose on Fedora 41
2. clone LibreChat repo
3. I modified .env and set `ANTHROPIC_API_KEY` as well as `ANTHROPIC_MODELS`
4. Run `docker compose up -d` - this executes successfully, though it gives warnings:
```
WARN[0000] The "UID" variable is not set. Defaulting to a blank string. 
WARN[0000] The "GID" variable is not set. Defaulting to a blank string. 
```
5. Web application is not runnning.

### What browsers are you seeing the problem on?

Firefox

### Relevant log output

```shell
$ docker logs LibreChat

Node.js v20.18.0

> LibreChat@v0.7.5 backend
> cross-env NODE_ENV=production node api/server/index.js

node:buffer:320
  throw new ERR_INVALID_ARG_TYPE(
  ^

TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined
    at Function.from (node:buffer:320:9)
    at Object.<anonymous> (/app/api/server/utils/crypto.js:4:20)
    at Module._compile (node:internal/modules/cjs/loader:1469:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1548:10)
    at Module.load (node:internal/modules/cjs/loader:1288:32)
    at Module._load (node:internal/modules/cjs/loader:1104:12)
    at Module.require (node:internal/modules/cjs/loader:1311:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (/app/api/server/utils/index.js:7:21)
    at Module._compile (node:internal/modules/cjs/loader:1469:14) {
  code: 'ERR_INVALID_ARG_TYPE'
}

$ docker ps
CONTAINER ID   IMAGE                                                   COMMAND                  CREATED          STATUS                          PORTS      NAMES
807fe96657b2   ghcr.io/danny-avila/librechat-dev:latest                "docker-entrypoint.s…"   25 minutes ago   Restarting (1) 46 seconds ago              LibreChat
d943b80c7e85   ghcr.io/danny-avila/librechat-rag-api-dev-lite:latest   "python main.py"         25 minutes ago   Up 16 minutes                              rag_api
75bab344aa2a   ankane/pgvector:latest                                  "docker-entrypoint.s…"   25 minutes ago   Up 16 minutes                   5432/tcp   vectordb
1ddb8788bde5   mongo                                                   "docker-entrypoint.s…"   25 minutes ago   Restarting (1) 15 seconds ago              chat-mongodb
393dfbc56fae   getmeili/meilisearch:v1.7.3                             "tini -- /bin/sh -c …"   25 minutes ago   Restarting (1) 15 seconds ago              chat-meilisearch
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Check your environment variables, do you have these set? 

https://www.librechat.ai/toolkit/creds_generator

**Reply by eldar** [reply]
Thanks for getting back to me! These variables are set with default values from the GitHub repo in the `.env` file that I copied from `.env.example` as per instructions. Should I generate the new values and put them to the `.env` file?

**Reply by eldar** [reply]
I updated the `.env` file with the newly generated keys, rebuilt the instance with docker compose, but am still getting the same error message.

**Reply by danny-avila** [reply]
The error comes from this line of code:

```cjs
const key = Buffer.from(process.env.CREDS_KEY, 'hex');
const iv = Buffer.from(process.env.CREDS_IV, 'hex');
```

how are you restarting? maybe the updated .env is not taking effect. Try `docker compose down` and then start it with `docker compose up`

**Reply by eldar** [reply]
I just did `docker compose down` followed by `docker compose up` and got the same errors in the output. Before these errors there also a bunch of errors looking like this:

```
chat-mongodb exited with code 0
chat-mongodb      | find: '/data/db': Permission denied
chat-mongodb      | chown: changing ownership of '/data/db': Permission denied
chat-meilisearch  | Error: Permission denied (os error 13)
chat-meilisearch  | Error: Permission denied (os error 13)
```

I also installed LibreChat on a different machine running OpenSUSE Tumbleweed, and everything is working flawlessly.

**Reply by uniq0de** [reply]
I have the same issue when I create a new image and run a container. I used the `Dockerfile.multi` to build the image and only replaced the image name of the `api` service in the `docker-compose.yaml` file. However, when I run the backend locally using the **same** `.env` file, no error appears. So, I am using the same `.env` file, but the same code inside the container fails.

**Reply by uniq0de** [reply]
@eldar the below helped me to resolve the issue. It seems that the .env is not being set correctly. 

services:
  api:
    container_name: LibreChat
    environment:
      - CREDS_KEY=xxx
      - CREDS_IV=xxx
      - etc.

**Reply by uniq0de** [reply]
and one more thing :) you may have made my mistake to bind the `.env` file to the `api` service. but it should be given as `env_file`. Then the above work around is not required anymore.

**Reply by j842** [reply]
Happened to me too - unraid environment. docker-compose up gave me:

```
TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined
at Function.from (node:buffer:320:9)

```

Fix was to add
```
    env_file:
      - .env
```

to the api service in the docker-compose.yml.
